### PR TITLE
Refactoring wagtail blocks

### LINKED ISSUE #1801: refactor wagtailpages blocks
they're currently all in one big file; it would be much nicer to have then housed in their own files inside a `blocks` dir.

### LINKED ISSUE #2362: Refactor wagtailpages/models.py into a folder
We've got models defined in files outside of `wagtailpages/models.py` that we need to import manually (and add linter ignore flags) in order to get django to be "aware" of them. This isn't ideal, so lets create a folder `wagtailpages/models` and define models there, so django can pick them up automatically without the need for janky imports.

### PR DESCRIPTION
**Pull request by youriwims**
Closes #1801, closes #2362

_Updates made based on this [comment](https://github.com/mozilla/foundation.mozilla.org/issues/1801#issuecomment-477252339)._

Test Page: https://foundation-mofostaging-pr-2999.herokuapp.com/en/refactor-test/ *
*_link updated 22/04/2019_

**Documentation:**
- [x] Is my code documented?
- [x] Did I update the READMEs or wagtail documentation?

### CODE REVIEW

**Comment by youriwims** [comment]
@Pomax & @mmmavis please let me know if there was anything that I may have missed. I tried to update the blocks documentation in the README a bit as well; let me know if anything should be reworded or changed. Thanks!

File: network-api/networkapi/wagtailpages/customblocks/__init__.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    'ProfileDirectory',
+    'PulseProjectList',
+    'QuoteBlock',
+    'VideoBlock'
+]
+
+# Deprecated/ Unused ?
+
+
+class FigureBlock2(blocks.StructBlock):
```
Comment (youriwims) [inline network-api/networkapi/wagtailpages/customblocks/__init__.py]: Filed a [follow-up ticket](https://github.com/mozilla/foundation.mozilla.org/issues/3003) to remove this.

File: network-api/networkapi/wagtailpages/customblocks/__init__.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        help_text='Optional URL that this figure should link out to.',
+    )
+    square_image = blocks.BooleanBlock(
+        default=True,
+        required=False,
+        help_text='If left checked, the image will be cropped to be square.'
+    )
+
+
+class AlignedImageBlock(ImageBlock):
```
Comment (youriwims) [inline network-api/networkapi/wagtailpages/customblocks/__init__.py]: @Pomax @mmmavis is this an older block?...it doens't appear that we're using this anywhere.

**Comment by mmmavis** [comment]
PR looks great to me! ✨  I wasn't very involved with the original wagtail blocks work and I am not a Django expert so I'm gonna remove myself as PR reviewer and let @Pomax have a thorough check instead. 🤓

File: network-api/networkapi/wagtailpages/customblocks/bootstrap_spacer.py [hunk 3]
```diff
@@ -0,0 +1,46 @@
+from wagtail.core import blocks
+
+
+class BootstrapSpacerBlock(blocks.StructBlock):
+    """
+    See https://getbootstrap.com/docs/4.0/utilities/spacing/
+    """
+
+    # property = blocks.ChoiceBlock(
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/customblocks/bootstrap_spacer.py]: we can probably remove L9-L15

File: network-api/networkapi/wagtailpages/customblocks/image_text_mini.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+from wagtail.images.blocks import ImageChooserBlock
+from wagtail.core import blocks
+
+
+class ImageBlock(blocks.StructBlock):
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/customblocks/image_text_mini.py]: triplicate code =)

File: network-api/networkapi/wagtailpages/customblocks/profile_directory.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+
+    @property
+    def rev(self):
+        # The default API behaviour is newest-first, so the "rev" attribute
+        # should only have an attribute value when oldest-first is needed.
+        newest_first = self['newest_first']
+        return True if newest_first else ''
+
+
+class LatestProfileList(blocks.StructBlock):
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/customblocks/profile_directory.py]: duplicate code: this already exists in `pulse-profile-list.py`

File: network-api/networkapi/wagtailpages/customblocks/pulse_project_list.py [hunk 6]
```diff
@@ -0,0 +1,205 @@
...
+
+    @property
+    def rev(self):
+        # The default API behaviour is newest-first, so the "rev" attribute
+        # should only have an attribute value when oldest-first is needed.
+        newest_first = self['newest_first']
+        return True if newest_first else ''
+
+
+class LatestProfileList(blocks.StructBlock):
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/customblocks/pulse_project_list.py]: we should put this in its own file, though.

File: network-api/networkapi/wagtailpages/customblocks/annotated_image_block.py [hunk 7]
```diff
@@ -0,0 +1,29 @@
+from wagtail.images.blocks import ImageChooserBlock
+from wagtail.core import blocks
+
+
+class ImageBlock(blocks.StructBlock):
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/customblocks/annotated_image_block.py]: This should be in its own `image_block.py`

File: network-api/networkapi/wagtailpages/customblocks/image_text.py [hunk 8]
```diff
@@ -0,0 +1,46 @@
+from wagtail.images.blocks import ImageChooserBlock
+from wagtail.core import blocks
+
+
+class ImageBlock(blocks.StructBlock):
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/customblocks/image_text.py]: duplicate code (and should be in its own file)

**Comment by Pomax** [comment]
turns out you can't just say "I meant to click request-changes, not comment"... thanks github

File: network-api/networkapi/wagtailpages/customblocks/__init__.py (hunk 2 above)
Comment (youriwims) [reply network-api/networkapi/wagtailpages/customblocks/__init__.py]: Follow up filed here ➡️ #3008

File: network-api/networkapi/wagtailpages/customblocks/__init__.py [hunk 9]
```diff
@@ -0,0 +1,73 @@
+from wagtail.images.blocks import ImageChooserBlock
+from wagtail.core import blocks
+
+from .air_table_block import AirTableBlock
+from .bootstrap_spacer import BootstrapSpacerBlock
+from .iframe_block import iFrameBlock
+from .annotated_image_block import AnnotatedImageBlock
+from .image_grid import ImageGrid, ImageGridBlock
+from .image_text_mini import ImageTextMini, ImageBlock
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/customblocks/__init__.py]: `ImageBlock` isn't reexported, so we don't need the import here.

```suggestion
from .image_text_mini import ImageTextMini
```

File: network-api/networkapi/wagtailpages/customblocks/__init__.py [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+from .image_text import ImageTextBlock
+from .latest_profile_list import LatestProfileQueryValue, LatestProfileList
+from .link_button import LinkButtonBlock
+from .profile_by_id import ProfileById
+from .profile_directory import ProfileDirectory
+from .pulse_project_list import PulseProjectList
+from .quote_block import QuoteBlock
+from .video_block import VideoBlock
+
+__all__ = [
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/customblocks/__init__.py]: Do we actually need `__all__`? We're just pass-through importing so there aren't any usually-hidden objects, or objects that need aliasing.

File: network-api/networkapi/wagtailpages/customblocks/__init__.py [hunk 11]
```diff
@@ -0,0 +1,73 @@
...
+    'LatestProfileList',
+    'LinkButtonBlock',
+    'ProfileById',
+    'ProfileDirectory',
+    'PulseProjectList',
+    'QuoteBlock',
+    'VideoBlock'
+]
+
+# Deprecated/ Unused ?
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/customblocks/__init__.py]: Let's put these in their own files anyway, just to keep `__init__.py` lean.

File: network-api/networkapi/wagtailpages/customblocks/image_grid.py [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+        help_text='Optional URL that this figure should link out to.',
+    )
+    square_image = blocks.BooleanBlock(
+        default=True,
+        required=False,
+        help_text='If left checked, the image will be cropped to be square.'
+    )
+
+
+class ImageGridBlock(blocks.StructBlock):
```
Comment (Pomax) [inline network-api/networkapi/wagtailpages/customblocks/image_grid.py]: let's put this in its own file. It's a small bit of code, but if we're doing this refactor, we might as well do it fully.

**Comment by Pomax** [comment]
We probably want to block this PR on a new issue/PR that makes sure that we cover all the components involved in this refactor in our tests. Then we can land that first, making sure the Percy has screenshots for all those test, and then we can rebase this PR on that so that percy can tell us whether any of the refactors "missed" something", or even broke something, in a way that ensures there won't be any regressions later.

File: network-api/networkapi/wagtailpages/customblocks/__init__.py (hunk 10 above)
Comment (youriwims) [reply network-api/networkapi/wagtailpages/customblocks/__init__.py]: cool, i'll remove it

Comment (youriwims) [reply network-api/networkapi/wagtailpages/customblocks/__init__.py]: @Pomax ⬆️

Comment (Pomax) [reply network-api/networkapi/wagtailpages/customblocks/__init__.py]: hm, reading up on flake8 ignore abilities, looks like `__all__` is the easiest "fix"... let's keep it in then!

**Comment by youriwims** [comment]
Blocked as per the previous comment ⬆️

**Comment by Pomax** [comment]
closing this PR in favour a redo of this work, where we're refactoring individual blocks per PR.

**Pull request closed without merging**